activerecord/lib/active_record/core.rb

```diff
@@ -354,6 +354,8 @@ def inspect # :nodoc:
           super
         elsif abstract_class?
           "#{super}(abstract)"
+        elsif singleton_class?
+          super
         elsif !schema_loaded? && !connected?
           "#{super} (call '#{super}.load_schema' to load schema informations)"
```

Comment:
As this failed previously when calling `table_exists?` should we move this new case just above `table_exists?`?
```suggestion
        elsif !schema_loaded? && !connected?
          "#{super} (call '#{super}.load_schema' to load schema informations)"
        elsif singleton_class?
          super
```